teps/0092-scheduling-timeout.md

```diff
@@ -17,361 +24,474 @@ authors:
-    taskruns: # this section would indicate that this timeout should be applied to all pipeline tasks
-      scheduling: "1m" # this is the new option
+    scheduling: "5m"
 ```
 
-When set:
-* This timeout would be applied to all `TaskRuns` created when executing the `Pipeline`
-* The order of timeout application would be:
-  1. At the start of the `Pipeline` execution, the `pipeline` and `tasks` timeout counters would begin
-  2. At the start of each `Task` being executed as a `TaskRun`, the `scheduling` timeout counter would begin,
-     considering the timeout to have started from the creation time of the `TaskRun` in etcd
-  3. Once the pod corresponding to the `TaskRun` has been scheduled, if the Pipeline Task configuration
-     [includes a timeout for the individual Task,](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#configuring-the-failure-timeout)
-     the counter for that timeout would begin
-  4. Once the `finally` Tasks in the `Pipeline` begin executing, the `tasks` timeout counter would no loner be counting,
-     and the `finally` counter would start
-  5. The `scheduling` timeout would apply to the `finally` Tasks in the same way as it is applied to the other tasks
+The spec that is calculated is this:
 
-When not set, no scheduling timeout would apply (i.e. there is no scheduling timeout by default).
+```yaml
+spec:
+  timeouts:
+    scheduling: "5m"  # User-specified value
+    execution: "55m"  # Calculated from default total timeout - scheduling timeout
+    total: "60m"  # Default value
+```
 
-### TaskRun
+###### User does not care when the `Pod` is scheduled but wants the `TaskRun` executed within a specific timeframe
 
-[TaskRun still currently only has one timeout field](https://github.com/tektoncd/pipeline/blob/main/docs/taskruns.md#configuring-the-failure-timeout)
-(while [`Pipeline` configuration was recently updated to make a dictionary for different timeout configuration](https://github.com/tektoncd/community/blob/main/teps/0046-finallytask-execution-post-timeout.md#proposal)):
+User creates this:
 
 ```yaml
 spec:
-  timeout: 60s
+  timeouts:
+    scheduling: "0m"
+    execution: "20m"
+    total: "0m"
+
 ```
 
-To support this new `scheduling` timeout we would follow
-[the same pattern we followed with adding more timeout configuration to PipelineRuns](https://github.com/tektoncd/community/blob/main/teps/0046-finallytask-execution-post-timeout.md)
-and add a dictionary to hold configuration for the existing timeout feature and this new one. The previous `timeout`
-field would be deprecated once this feature is beta.
+The spec that is calculated is this:
 
 ```yaml
 spec:
   timeouts:
-    scheduling: "5m" # this is the new option
-    execution: "10m" # this option would have the same functionality as the existing `timeout` field
+    scheduling: "0m"  # User-specified value
+    execution: "20m"  # User-specified value
+    total: "20m"   # Calculated from scheduling + execution timeout
```

Comment:
this is confusing, I'd have expected the total to always remain "0m" which means it'd run until completion (no timeout) -- cc @EmmaMunley @lbernick